radix-engine-interface/src/blueprints/utils.rs

```diff
@@ -0,0 +1,99 @@
+/// appropriate [`BlueprintId`].
+pub fn resolve_main_module_blueprint_id(entity_type: EntityType) -> Option<&'static BlueprintId> {
+    match entity_type {
+        EntityType::GlobalPackage => Some(package_blueprint_id()),
+        EntityType::GlobalConsensusManager => Some(consensus_manager_blueprint_id()),
+        EntityType::GlobalValidator => Some(validator_blueprint_id()),
+        EntityType::GlobalTransactionTracker => Some(transaction_tracker_blueprint_id()),
+        EntityType::GlobalAccessController => Some(access_controller_blueprint_id()),
+        EntityType::GlobalOneResourcePool => Some(one_resource_pool_blueprint_id()),
+        EntityType::GlobalTwoResourcePool => Some(two_resource_pool_blueprint_id()),
+        EntityType::GlobalMultiResourcePool => Some(multi_resource_pool_blueprint_id()),
+        EntityType::GlobalAccountLocker => Some(account_locker_blueprint_id()),
+        EntityType::GlobalFungibleResourceManager => Some(fungible_resource_manager_blueprint_id()),
+        EntityType::GlobalNonFungibleResourceManager => {
+            Some(non_fungible_resource_manager_blueprint_id())
+        }
+        EntityType::InternalFungibleVault => Some(fungible_vault_blueprint_id()),
+        EntityType::InternalNonFungibleVault => Some(non_fungible_vault_blueprint_id()),
+        EntityType::GlobalAccount
+        | EntityType::GlobalPreallocatedSecp256k1Account
+        | EntityType::GlobalPreallocatedEd25519Account => Some(account_blueprint_id()),
+        EntityType::GlobalIdentity
+        | EntityType::GlobalPreallocatedSecp256k1Identity
+        | EntityType::GlobalPreallocatedEd25519Identity => Some(identity_blueprint_id()),
+        EntityType::InternalGenericComponent
+        | EntityType::InternalKeyValueStore
+        | EntityType::GlobalGenericComponent => None,
+    }
+}
+
+/// This function resolves the [`BlueprintId`] of the invoked blueprint given the [`BlueprintId`] of
+/// the main module and the [`ModuleId`] being invoked.
+pub fn resolve_invoked_blueprint_id(
+    main_module_blueprint_id: &BlueprintId,
+    module_id: ModuleId,
+) -> Option<&BlueprintId> {
+    match module_id {
+        ModuleId::Main => Some(main_module_blueprint_id),
+        // TODO: We could improve this to take into account if the blueprint has these modules or
+        // not. For the time being this doesn't seem to be needed.
+        ModuleId::Metadata => Some(metadata_blueprint_id()),
+        ModuleId::Royalty => Some(royalty_blueprint_id()),
+        ModuleId::RoleAssignment => Some(role_assignment_blueprint_id()),
+    }
+}
+
+macro_rules! define_static_blueprint_id {
+    (
+        $(
+            $name: ident => ($package_address: expr, $blueprint_name: expr)
+        ),* $(,)?
+    ) => {
+        paste::paste! {
+            $(
+                pub fn [< $name _blueprint_id >]() -> &'static BlueprintId {
+                    static BLUEPRINT_ID: std::sync::OnceLock<crate::prelude::BlueprintId> =
```

Comment:
This won't work for no-std. We use `lazy_static` for that reason.

Reply:
Or honestly, it's not a big deal to me if `resolve_main_module_blueprint_id` just returns an owned blueprint id, or returns a `(PackageAddress, &'static str)`.

Reply:
I ended up using lazy static here so we don't have no-std issues.